src/shared.rs

```diff
@@ -1,24 +1,18 @@
-use serde::de::DeserializeOwned;
-use serde::Serialize;
-use std::collections::HashMap;
-use std::env::VarError;
-use std::error::Error;
-use std::ffi::{OsStr, OsString};
-use std::fmt::Debug;
-use std::fmt::Display;
-use std::fs;
-use std::io;
-use std::path::{Iter, Path};
+use crate::Error as CrateError;
+use serde::{de::DeserializeOwned, Serialize};
+use std::{error::Error, fmt::Display, fs, path::Path};
 
-pub fn write_toml_file(value: &impl Serialize, path: impl AsRef<Path>) -> io::Result<()> {
+pub fn write_toml_file(value: &impl Serialize, path: impl AsRef<Path>) -> Result<(), CrateError> {
     // TODO: Fix Result type, remove unwrap
```

Comment:
It looks like those TODO comments can go now. :)

Reply:
good catch!